hooks/useFetchNodeReleasesData.ts

```diff
@@ -0,0 +1,47 @@
+import { useRouter } from 'next/router';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { getNodeReleaseStatus } from '../util/nodeReleaseData';
+import type { NodeReleaseData } from '../types';
+
+const fetcher = (...args: Parameters<typeof fetch>) =>
+  fetch(...args).then(res => res.json());
+
+export const useFetchNodeReleasesData = (): NodeReleaseData[] => {
+  const { basePath } = useRouter();
+
+  const { data = [] } = useSWR<any[]>(
```

Comment:
I'm not keen of using the any type here. Because then down there we don't have types for the `data.map`.

I think you can get the type of this JSON file without importing it by doing:

```tsx
import type NodeReleaseDataResponse from '../public/static/node-releases-data.json';
```

Can you give a try?

Reply:
Otherwise, we can create this type manually, I guess... (Directly on the top of this hook as `type NodeReleasesDataResponse`

Reply:
This works:
```diff
@@ -2,6 +2,7 @@ import { useRouter } from 'next/router';
 import { useMemo } from 'react';
 import useSWR from 'swr';
 import { getNodeReleaseStatus } from '../util/nodeReleaseData';
+import type NodeReleasesDataResponse from '../public/static/node-releases-data.json';
 import type { NodeReleaseData } from '../types';
 
 const fetcher = (...args: Parameters<typeof fetch>) =>
@@ -10,7 +11,7 @@ const fetcher = (...args: Parameters<typeof fetch>) =>
 export const useFetchNodeReleasesData = (): NodeReleaseData[] => {
   const { basePath } = useRouter();
 
-  const { data = [] } = useSWR<any[]>(
+  const { data = [] } = useSWR<typeof NodeReleasesDataResponse>(
     `${basePath}/static/node-releases-data.json`,
     fetcher
   );
```
I tried it and it seems to behave correctly (no duplicate fetch) and no lint error
However, the syntax seems suspicious to me 🤔 
Let me search around

Reply:
But why do you need to add `typeof NodeReleasesDataResponse` if you're already importing a type?

Reply:
Also, it would be great to use it to analyze the bundle to see if it is not bundling the JSON file... Because that looks odd... A good way to analyze the bundle is on the Chrome Dev Tools > Source and then hit Search (as in the screenshot). Then search for any line or word that exists in the JSON file

![image](https://github.com/nodejs/nodejs.org/assets/12037269/8b8c8231-54f3-4339-930d-fa00d8b24bb9)

If after hitting enter you find nothing, then we're good. Otherwise, we need another way

Reply:
Actually I've tried:

1. This one gives lint error
```diff
@@ -2,6 +2,7 @@ import { useRouter } from 'next/router';
 import { useMemo } from 'react';
 import useSWR from 'swr';
 import { getNodeReleaseStatus } from '../util/nodeReleaseData';
+import NodeReleasesDataResponse from '../public/static/node-releases-data.json';
 import type { NodeReleaseData } from '../types';
 
 const fetcher = (...args: Parameters<typeof fetch>) =>
@@ -10,7 +11,7 @@ const fetcher = (...args: Parameters<typeof fetch>) =>
 export const useFetchNodeReleasesData = (): NodeReleaseData[] => {
   const { basePath } = useRouter();
 
-  const { data = [] } = useSWR<any[]>(
+  const { data = [] } = useSWR<typeof NodeReleasesDataResponse>(
     `${basePath}/static/node-releases-data.json`,
     fetcher
   );
```
```
5:1  error  All imports in the declaration are only used as types. Use `import type`  @typescript-eslint/consistent-type-imports
```

2. This one gives straight forward error
```diff
@@ -2,6 +2,7 @@ import { useRouter } from 'next/router';
 import { useMemo } from 'react';
 import useSWR from 'swr';
 import { getNodeReleaseStatus } from '../util/nodeReleaseData';
+import type NodeReleasesDataResponse from '../public/static/node-releases-data.json';
 import type { NodeReleaseData } from '../types';
 
 const fetcher = (...args: Parameters<typeof fetch>) =>
@@ -10,7 +11,7 @@ const fetcher = (...args: Parameters<typeof fetch>) =>
 export const useFetchNodeReleasesData = (): NodeReleaseData[] => {
   const { basePath } = useRouter();
 
-  const { data = [] } = useSWR<any[]>(
+  const { data = [] } = useSWR<NodeReleasesDataResponse>(
     `${basePath}/static/node-releases-data.json`,
     fetcher
   );
```
```
'NodeReleasesDataResponse' refers to a value, but is being used as a type here. Did you mean 'typeof NodeReleasesDataResponse'?ts(2749)
```

Reply:
Gotcha, this kinda makes me believe it is importing the JSON within the bundle.

I think it's worth checking if the JSON is being added to the bundle... If you have inconclusive results I can also try to check that by myself 🤔

Reply:
> import type only imports declarations to be used for type annotations and declarations. It always gets fully erased

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html
TS docs said that `import type` is omitted. I tried it myself as well and it wasn't included in the bundle.

> TS will be unhappy

Yeah, I just realized it yesterday as well after I tried deleting the JSON file 😅 

Normally, the data structure is shared by Server & Client but, I think it doesn't worth to go through such length for this case (overcomplicated).

Making the type on top of this file sounds good, thanks!

Reply:
I will after addressing this! 🙇